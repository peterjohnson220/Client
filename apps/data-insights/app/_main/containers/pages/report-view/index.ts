export * from './report-view.page';
