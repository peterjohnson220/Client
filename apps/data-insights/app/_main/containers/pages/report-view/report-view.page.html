<ng-container *ngIf="(workbookViewUrl$ | async) as workbookViewUrlAsync">
  <pf-page-with-title-bar [returnUrl]="'/'">
    <ng-container page-title>{{workbookTitle}}</ng-container>
    <ng-container page-content>
      <pf-async-container
        [loading]="workbookViewUrlAsync.loading"
        [loadingError]="workbookViewUrlAsync.loadingError"
        [loadingErrorMessage]="'Error loading this workbook'"
        [hideReloadButton]="true">
        <div id="vizContainer">

        </div>
      </pf-async-container>
    </ng-container>
  </pf-page-with-title-bar>
</ng-container>
