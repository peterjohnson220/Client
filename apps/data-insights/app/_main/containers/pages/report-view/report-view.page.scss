@import 'colors';
@import 'data-insights-variables';

.vizContainer {
  width: 100%;
}
