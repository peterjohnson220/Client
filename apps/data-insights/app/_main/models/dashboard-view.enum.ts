export enum DashboardView {
  All = 'All Dashboards',
  Favorites = 'Favorites',
  Views = 'All Views'
}
