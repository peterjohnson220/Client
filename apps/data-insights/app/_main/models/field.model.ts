export interface Field {
  EntityId: number;
  Entity: string;
  EntitySourceName: string;
  DataElementId: number;
  SourceName: string;
  DisplayName: string;
  KendoGridField?: string;
  DataType?: FieldDataType;
  IsSelected?: boolean;
  IsSortable: boolean;
  Order?: number;
  DataElementOrder?: number;
  FormulaId?: number;
  FieldType: FieldType;
  Format?: string;
  IsEditable?: boolean;
  Formula?: string;
  FormulaName?: string;
}

export interface FieldListItem {
  Entity: string;
  IsSelected: boolean;
  DisplayName: string;
  FieldListItemId: string;
}

export enum FieldDataType {
  Date = 'Date',
  Int = 'int',
  Float = 'float',
  String = 'String',
  LongString = 'longString',
  Bit = 'bit'
}

export enum FieldType {
  DataElement = 'DataElement',
  Formula = 'Formula'
}

export function generateMockField(): Field {
  return {
    DataElementId: 1,
    EntityId: 1,
    Entity: 'Jobs',
    DataType: FieldDataType.String,
    DisplayName: 'Job Title',
    EntitySourceName: 'CompanyJobs',
    IsSelected: false,
    Order: 1,
    SourceName: 'Job_Title',
    KendoGridField: 'CompanyJobs.Job_Title',
    IsSortable: true,
    FieldType: FieldType.DataElement,
    Format: 'N1'
  };
}
