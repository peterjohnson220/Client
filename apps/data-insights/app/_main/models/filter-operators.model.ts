export interface FilterOperator {
  Name: string;
  Value: string;
}

export const Equals: FilterOperator = {
  Name: 'equals',
  Value: 'in'
};

export const Between: FilterOperator = {
  Name: 'between',
  Value: 'between'
};

export const GreaterThan: FilterOperator = {
  Name: 'greater than',
  Value: '>'
};

export const GreaterThanOrEqual: FilterOperator = {
  Name: 'greater than or equal',
  Value: '>='
};

export const LessThan: FilterOperator = {
  Name: 'less than',
  Value: '<'
};

export const LessThanOrEqual: FilterOperator = {
  Name: 'less than or equal',
  Value: '<='
};

export const IsBefore: FilterOperator = {
  Name: 'is before',
  Value: '<'
};

export const IsAfter: FilterOperator = {
  Name: 'is after',
  Value: 'isafterdate'
};

export const Is: FilterOperator = {
  Name: 'is',
  Value: 'isdate'
};

export const Contains: FilterOperator = {
  Name: 'contains',
  Value: 'contains'
};
