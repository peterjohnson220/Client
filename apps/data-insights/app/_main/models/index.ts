export * from './dashboard-view.enum';
export * from './workbook.model';
export * from './save-workbook-tag-obj.model';
export * from './report-view-types.enum';
export * from './view.model';
export * from './entity.model';
export * from './save-user-workbook-modal-data.model';
export * from './user-data-view.model';
export * from './report-types.enum';
export * from './field.model';
export * from './save-workbook-mode.enum';
export * from './save-report-order-data.model';
