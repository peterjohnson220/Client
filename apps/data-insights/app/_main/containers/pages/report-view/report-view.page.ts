import { Component } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

import { ReportViewTypes } from 'libs/features/surveys/reports/models';

@Component({
  selector: 'pf-report-view-page',
  templateUrl: './report-view.page.html',
  styleUrls: ['./report-view.page.scss']
})
export class ReportViewPageComponent {
  reportTitle: string;
  workbookId: string;
  workbookName: string;
  viewName: string;
  viewType: ReportViewTypes;

  constructor(private route: ActivatedRoute) {
    this.reportTitle = this.route.snapshot.queryParamMap.get('title');
    this.workbookId = this.route.snapshot.params.workbookId;
    this.viewName = this.route.snapshot.params.viewName;
    this.viewType = this.route.snapshot.data.viewType;
    this.workbookName = this.route.snapshot.params.workbookName;
  }
}
