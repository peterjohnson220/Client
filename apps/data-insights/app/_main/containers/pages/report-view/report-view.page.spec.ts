import { TestBed, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

import { ReportViewTypes } from 'libs/features/reports/models';

import { ReportViewPageComponent } from './report-view.page';

describe('Data Insights - Report View Page Component', () => {
  let instance: ReportViewPageComponent;
  let fixture: ComponentFixture<ReportViewPageComponent>;
  let route: ActivatedRoute;
  const queryStringParams = { 'title': 'Salary Structures', 'showTabs' : 'true' };

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [ ReportViewPageComponent ],
      schemas: [ NO_ERRORS_SCHEMA ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: {
            snapshot: {
              queryParamMap: { get: (key) =>  queryStringParams[key] },
              params: { viewName: 'Test View Name', workbookName: 'Test Workbook Name', workbookId: '123' },
              data: { viewType: ReportViewTypes.StandardWorkbook }
            }
          }
        }
      ]
    });

    fixture = TestBed.createComponent(ReportViewPageComponent);
    instance = fixture.componentInstance;
    route = TestBed.inject(ActivatedRoute);

    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
