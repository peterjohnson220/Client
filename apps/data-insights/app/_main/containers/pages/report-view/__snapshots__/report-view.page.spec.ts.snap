// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Insights - Report View Page Component should match snapshot 1`] = `
<pf-report-view-page
  __ngContext__={[Function LRootView]}
  reportTitle={[Function String]}
  route={[Function Object]}
  viewName={[Function String]}
  viewType="0"
  workbookId={[Function String]}
  workbookName={[Function String]}
>
  <pf-page-with-title-bar>
    Salary Structures
    <pf-report-view />
  </pf-page-with-title-bar>
</pf-report-view-page>
`;
